src/actions/LoginActions.tsx

```diff
@@ -73,22 +73,28 @@ export function initializeAccount(navigation: NavigationBase, account: EdgeAccou
       const fiatCurrencyCode = 'iso:' + defaultFiat
 
       const newAccountFlow = async (navigation: NavigationProp<'createWalletSelectCrypto'>, items: WalletCreateItem[]) => {
-        // New user FIO handle registration flow (if env is properly configured)
-        const { freeRegApiToken = '', freeRegRefCode = '' } = typeof ENV.FIO_INIT === 'object' ? ENV.FIO_INIT : {}
-        if (freeRegApiToken !== '' && freeRegRefCode !== '') {
-          Airship.show<boolean>(bridge => <FioCreateHandleModal bridge={bridge} />).then(isCreateHandle => {
-            if (isCreateHandle) navigation.navigate('fioCreateHandle', { freeRegApiToken, freeRegRefCode })
-          })
-        }
-
         navigation.replace('edgeTabs', {
           screen: 'walletsTab',
           params: {
             screen: 'walletList'
           }
         })
         const selectedEdgetokenIds = items.map(item => ({ pluginId: item.pluginId, tokenId: item.tokenId }))
-        await createCustomWallets(account, fiatCurrencyCode, selectedEdgetokenIds, dispatch)
+
+        // New user FIO handle registration flow (if env is properly configured)
+        const { freeRegApiToken = '', freeRegRefCode = '' } = typeof ENV.FIO_INIT === 'object' ? ENV.FIO_INIT : {}
+        const isCreateHandle =
+          freeRegApiToken !== '' && freeRegRefCode !== '' && (await Airship.show<boolean>(bridge => <FioCreateHandleModal bridge={bridge} />))
+
+        if (isCreateHandle) {
+          // We need to start the wallet creation process before starting the
+          // FIO workflow or wallet creation can fail.
+          createCustomWallets(account, fiatCurrencyCode, selectedEdgetokenIds, dispatch)
```

Comment:
Definitely want to await this before calling updateWalletsRequest()

Reply:
Commit removed